common/arg.cpp

```diff
@@ -699,13 +699,15 @@ gpt_params_context gpt_params_parser_init(gpt_params & params, llama_example ex,
             params.n_chunks = value;
         }
     ).set_examples({LLAMA_EXAMPLE_IMATRIX, LLAMA_EXAMPLE_PERPLEXITY, LLAMA_EXAMPLE_RETRIEVAL}));
+#ifdef FLASH_ATTN_AVAILABLE
     add_opt(llama_arg(
         {"-fa", "--flash-attn"},
         format("enable Flash Attention (default: %s)", params.flash_attn ? "enabled" : "disabled"),
         [](gpt_params & params) {
             params.flash_attn = true;
         }
     ).set_env("LLAMA_ARG_FLASH_ATTN"));
+#endif // FLASH_ATTN_AVAILABLE
```

Comment:
AFAIK flash attention is a safe option, meaning it will be disabled automatically if not supported. Removing this on non-supported build will be a breaking change.

Instead, print a message:

```suggestion
    add_opt(llama_arg(
        {"-fa", "--flash-attn"},
        format("enable Flash Attention (default: %s)", params.flash_attn ? "enabled" : "disabled"),
        [](gpt_params & params) {
            params.flash_attn = true;
#nifdef FLASH_ATTN_AVAILABLE
            fprintf(stderr, "warning: flash attention is not supported\n");
#endif // FLASH_ATTN_AVAILABLE
        }
    ).set_env("LLAMA_ARG_FLASH_ATTN"));
```

Reply:
Thanks for the review! I'll make the updates accordingly.